packages/headless/src/api/search/search-api-client.ts

```diff
@@ -36,9 +36,12 @@ import {SearchThunkExtraArguments} from '../../app/headless-engine';
 
 export type AllSearchAPIResponse = Plan | Search | QuerySuggest;
 
-export interface AsyncThunkSearchOptions<T extends Partial<SearchAppState>> {
+export interface AsyncThunkSearchOptions<
+  T extends Partial<SearchAppState>,
+  E = SearchAPIErrorWithStatusCode
+> {
   state: T;
-  rejectValue: SearchAPIErrorWithStatusCode;
+  rejectValue: E;
   extra: SearchThunkExtraArguments;
 }
```

Comment:
Discussed [here](https://github.com/coveo/ui-kit/pull/791#discussion_r628435328).

> This change is needed because the "show more" functionality triggers a search request on the collection we intend to expand.
>
> When the async thunk that sends the search request is rejected, we need to know which collection's "show more" request was rejected.